### PR TITLE
External accounts and provider connection flow

### LINKED ISSUE #5: APIs for provider accounts
Implement delete provider account and connect external account using delegated APIs.

See https://github.com/CareEvolution/Consumers/issues/9335, https://github.com/CareEvolution/Consumers/pull/9736, and https://github.com/CareEvolution/MyDataHelps.js/pull/17

### PR DESCRIPTION
**Pull request by mmertsock**
## Overview

Closes #5 

Implements APIs:

- Search external account providers
- Provider connection flow based on SFSafariViewController
- List, refresh, delete connected external accounts

The example app implements fully functional demos of all the above APIs. I also went back through the example code and added comments to better explain how it works and suggested ways to customize it for learning the SDK.

Note that the provider connection flow requires the app using this SDK to present and dismiss the SFSafariViewController. I didn't see any point in the SDK itself doing this, for a few reasons:

- It really only saves maybe one line of code
- The calling app may want to customize the Safari view (e.g. bar tint colors)
- The calling app can optionally implement SFSafariViewControllerDelegate to be notified if the user manually dismisses the view (cancelling the authorization flow)
- The precise mechanics of presenting and dismissing the Safari view depends on the UI framework (UIKit vs SwiftUI)
- The app must be responsible for intercepting the special link that dismisses the Safari view, either via a custom scheme or Universal Link. This involves entitlements/Info.plist configuration, and AppDelegate or SwiftUI code to accept the incoming link. It's not really possible, and I think inappropriate, for the SDK to do any of this, and really it's quite simple for the calling app to handle it. The example app code in this PR demonstrates how to do all of this, and I tried to document it thoroughly

Note that the example app uses a custom scheme implementation instead of Universal Links. This avoids registering a Bundle ID and team for the example app, setting up a domain, etc., and thus the example app remains easy for any developer to just build and run in Xcode.

This functionality is currently available on a test server, DM me for details on testing.

I added detailed documentation on the provider connection flow in `ExternalAccountProviders.swift`, I would appreciate review of this documentation text. Note that this is technical documentation specific to the iOS SDK and would be supplemental to a larger end-to-end provider accounts implementation guide.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc. Of particular note:
    - No temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
    - Xcode project/target settings should remain generic. Don't leak team identifiers, code signing info, local filesystem paths, etc.
- [x] My changes do not introduce any security risks, or any such risks have been properly mitigated

Security notes:

- Connects to some new delegated API endpoints. The usual security characteristics of delegated APIs apply, and I don’t see this introducing new security concerns for this SDK
- These new APIs all have simple requests and responses. Mostly basic integers, strings, dates. Text values are provider names/categories, nothing sensitive. And provider IDs are not sensitive values
- The most interesting API data is the special URL returned by the server that the SDK client then presents in a web view for the provider connection flow. The SDK and client app just consumes this URL, and it’s participant-specific and I think it’s ephemeral
- This SDK does not itself present the web view for the provider connection flow; the client app is responsible for this, and there’s a sample implementation here in the example app. The example/recommended implementation presents it via an SFSafariViewController, which is secure and private: it runs in a separate process and hides all user credentials/input and web content from the containing app

## Checklist

- [x] All public symbols are documented using Swift Markup comments
- [x] Source code file header comments are clean and standardized. Use "Created by CareEvolution on m/dd/yy"
- [x] Test and update the example app as needed. The example app should demonstrate all features of the SDK, and it's the most convenient way to test your feature

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

### CODE REVIEW

**Comment by chrisnowak** [comment]
Wow - this is awesome!

```
Note that the example app uses a custom scheme implementation instead of Universal Links. 
```

Definitely agree with this...seems vastly easier to drop in vs. having to manage universal links

**Comment by mmertsock** [comment]
Also, I noticed that iOS doesn't show the "Open in <app name>?" prompt when using the custom scheme in this particular use case, so it's not affected by one of the main UX drawbacks of custom schemes. Presumably the prompt doesn't appear because the app being opened is already in the foreground.

File: MyDataHelpsKit/ProviderConnections/ExternalAccountProviders.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    ///
+    /// Contact [CareEvolution support](https://developer.rkstudio.careevolution.com/help.html) to have your `finalRedirectURL` added to the list of allowed URLs in RKStudio. If the URL is not in the allow list, this API will produce an error result.
+    /// - Parameters:
+    ///   - provider: The external account provider to connect.
+    ///   - finalRedirectURL: A URL that is configured to open your app via a custom scheme or Universal Link.
+    ///   - completion: Called when the request is complete, with the provider authorization URL and supporting information on success, or an error on failure.
+    func connectExternalAccount(provider: ExternalAccountProvider, finalRedirectURL: URL, completion: @escaping (Result<ExternalAccountAuthorization, MyDataHelpsError>) -> Void) {
+        load(resource: ConnectExternalAccountResource(providerID: provider.id, finalRedirectURL: finalRedirectURL)) {
+            completion($0.map {
+                ExternalAccountAuthorization(provider: provider, authorizationURL: $0, finalRedirectURL: finalRedirectURL)
```
Comment (overcyn) [inline MyDataHelpsKit/ProviderConnections/ExternalAccountProviders.swift:39]: Do we need to pass the provider and finalRedirectURL parameters back into the completion? The caller has those values already.

**Review by overcyn — APPROVED** [review]
Code looks really good.

File: MyDataHelpsKit/ProviderConnections/ExternalAccountProviders.swift (hunk 1 above)
Comment (mmertsock) [reply MyDataHelpsKit/ProviderConnections/ExternalAccountProviders.swift:39]: Not strictly necessary, I just think it makes things easier for the caller to have everything in a single result object, given the asynchronous flow.

**Pull request merged**